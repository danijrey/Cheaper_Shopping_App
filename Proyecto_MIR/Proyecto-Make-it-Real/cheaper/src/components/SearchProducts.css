.App {
    color: black;
    background-color:white ;
    margin: 10px;
    align-items: center;
  }
  .title-search{
      margin :1%;
      padding: 5%;
      align-items: center;
      color:#e0c408c9 
  } 
  .list {
    display: grid;
    grid-template-columns: 130px 130px;
    grid-template-rows: 245px 245px 245px;
    grid-gap: 10px;
    justify-content: center;
    margin: 3%

  }
  
  .p-Productos{
    color: white;
    font-weight: 400;
    margin-bottom: 5px;
  }
  
  .H3-Productos {
    color: white;
    font-weight: bold;
    font-size: 24px;
    margin: 0px;
  }
  
  .VerMas {
    border-radius: 5px;
    font-weight: 500;
  }
  
  .VerMas:hover{
    background: rgba(158, 151, 151, 0.829)
  }
  
  
  .card { 
    height: 245px;
    width: 130px;
    align-items: center;
    
  }

  .cards:hover{
      transform: translateY(-8px);
      box-shadow: 0 12px 16px rgb(0, 0, 0, 0.2);
  }
  
  .row-search {
    height: 245px;
    width: 130px; 
    display: flex;
    flex-wrap: wrap;
  

  }
  
  .Imagenes {
    width: 120px;
    height: 120px;  
    margin-top: 8px; 
    border-radius: 5px;
  }
  
  .footer{
	background-color: rgba(128, 128, 128, 0.514);
	margin-bottom: 0;
	height:130px;
	padding-top: 10%;
}