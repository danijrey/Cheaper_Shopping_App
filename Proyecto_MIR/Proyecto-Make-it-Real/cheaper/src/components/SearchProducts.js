import React, { useState, useEffect } from 'react';
import logo from "../assets/logo.png";
import './SearchProducts.css';
import NavBar from './NavBar.js'
import { Link } from "react-router-dom"
import axios from "axios";


class SearchProducts extends React.Component {
  state = {
    name: "",
    products: []
  };


  componentDidMount() {
    console.log(this.props)

    const { products, ...data } = this.props.match.params

    axios({
      method: "POST",
      baseURL: process.env.REACT_APP_SERVER_URL,
      url: `/providers/search/${this.props.match.params.name}`,
      data,
      headers: {
        "Content-Type": "application/json",
      },
    })
      .then(response => {
        this.setState({ products: response.data })
        console.log(response.data);
      })
      .catch(error => {
        // history.push('/');
      })
  }

  componentDidUpdate() {
    console.log(this.props)

    const { products, ...data } = this.props.match.params

    axios({
      method: "POST",
      baseURL: process.env.REACT_APP_SERVER_URL,
      url: `/providers/search/${this.props.match.params.name}`,
      data,
      headers: {
        "Content-Type": "application/json",
      },
    })
      .then(response => {
        this.setState({ products: response.data })
        console.log(response.data);
      })
      .catch(error => {
        // history.push('/');
      })
  }

  render() {

    const { products } = this.state;

    return (
      <div className="ProductSearch" key={this.props.match.params.name}>
        <NavBar />

        <h5 className="title-search">Te damos las mejores opciones</h5>
        <div className="gridContainer">
          {products.map((data) => {
            return (
              <div className="row-List">

                <div className="cards" key={data._id}>
                  <img
                    src={data.picture}
                    className="Imagenes"
                    alt={data.picture} />
                  <h3 className="H3-Productos">
                    {data.name}
                  </h3>
                  <p className="p-Productos">
                    {data.description}
                  </p>
                  <br />
                  <Link to={`/providers/products/show/${data._id}`} className="VerMas" >Ver m√°s</Link>
                </div>
              </div>
            );
          })}
        </div>
        <footer className="footer">
          <img src={logo} className="NavBar-Logo-Home" alt="Logo"></img>
          <p className="copyright"> <span dangerouslySetInnerHTML={{ "__html": "&copy;" }} /> All rights reserved 2020 </p>
        </footer>

      </div>

    )
  }
}


export default SearchProducts;