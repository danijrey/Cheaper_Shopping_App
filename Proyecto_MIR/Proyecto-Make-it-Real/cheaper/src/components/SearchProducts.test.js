import React from 'react';
import {render} from '@testing-library/react';
import SearchProducts from './SearchProducts';


describe ('SearchProducts',() =>{
    it ('should render correctly', ()=>{
    const {container} = render (<SearchProducts/>); 
    expect (container.firstChild).toMatchSnapshot();
    });
});