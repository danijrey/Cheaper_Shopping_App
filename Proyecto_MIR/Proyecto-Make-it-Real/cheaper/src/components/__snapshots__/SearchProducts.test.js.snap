// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchProducts should render correctly 1`] = `
<div
  class="ProductSearch"
>
  <header
    class="toolbar"
  >
    <nav
      class="toolbar-navigation"
    >
      <div
        class="toolbar-toggle-button"
      >
        <button
          class="toggle-button"
        >
          <div
            class="toggle-button_line"
          />
          <div
            class="toggle-button_line"
          />
          <div
            class="toggle-button_line"
          />
        </button>
      </div>
      <img
        alt="Logo"
        class="toolbar-logo"
        src="logo.png"
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-shopping-cart fa-w-18 NavBar-Icons"
        data-icon="shopping-cart"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
          fill="currentColor"
        />
      </svg>
    </nav>
  </header>
  <nav
    class="side-drawer"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Mi perfil
        </a>
      </li>
      <li>
        <a
          href="/products/all"
        >
          Mis productos
        </a>
      </li>
      <li>
        <a
          href="/home"
        >
          Inicio
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Configuracion
        </a>
      </li>
      <li>
        <a
          href="/providers/:id/branch/create"
        >
          Crear sucursal
        </a>
      </li>
      <img
        alt="Logo"
        class="side-drawer-logo"
        src="logo.png"
      />
    </ul>
  </nav>
  <div
    class="NavBusqueda"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-search fa-w-16 NavBusqueda-Icon"
      data-icon="search"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        fill="currentColor"
      />
    </svg>
    <input
      class="NavBusqueda-Busqueda"
      placeholder=" Realizar una busqueda."
      type="text"
    />
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-camera fa-w-16 NavBusqueda-Icon"
      data-icon="camera"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="Ubicacion"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-map-marker-alt fa-w-12 Ubicacion-Icon"
      data-icon="map-marker-alt"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
        fill="currentColor"
      />
    </svg>
    <input
      class="Ubicacion-Busqueda"
      placeholder=" Ubicación."
      type="text"
    />
  </div>
  <h5
    class="title-search"
  >
    Te damos las mejores opciones
  </h5>
  <div
    class="list"
  >
    <div
      class="row-search"
    >
      <div
        class="cards"
      >
        <img
          alt="Panadería"
          class="Panadería"
          src="panaderia.jpg"
        />
        <h3
          class="H3-Productos"
        >
          Panadería
        </h3>
        <p
          class="p-Productos"
        >
          Lorem ipsum dolor sit amet
        </p>
        <button
          class="VerMas"
        >
          Ver más
        </button>
      </div>
    </div>
    <div
      class="row-search"
    >
      <div
        class="cards"
      >
        <img
          alt="Lácteos"
          class="Lácteos"
          src="lacteos.jpg"
        />
        <h3
          class="H3-Productos"
        >
          Lácteos
        </h3>
        <p
          class="p-Productos"
        >
          Lorem ipsum dolor sit amet
        </p>
        <button
          class="VerMas"
        >
          Ver más
        </button>
      </div>
    </div>
    <div
      class="row-search"
    >
      <div
        class="cards"
      >
        <img
          alt="Verduras"
          class="Verduras"
          src="verduras.jpg"
        />
        <h3
          class="H3-Productos"
        >
          Verduras
        </h3>
        <p
          class="p-Productos"
        >
          Lorem ipsum dolor sit amet
        </p>
        <button
          class="VerMas"
        >
          Ver más
        </button>
      </div>
    </div>
    <div
      class="row-search"
    >
      <div
        class="cards"
      >
        <img
          alt="Carnes"
          class="Carnes"
          src="carnes.jpg"
        />
        <h3
          class="H3-Productos"
        >
          Carnes
        </h3>
        <p
          class="p-Productos"
        >
          Lorem ipsum dolor sit amet
        </p>
        <button
          class="VerMas"
        >
          Ver más
        </button>
      </div>
    </div>
    <div
      class="row-search"
    >
      <div
        class="cards"
      >
        <img
          alt="Abarrotes"
          class="Abarrotes"
          src="abarrotes.jpg"
        />
        <h3
          class="H3-Productos"
        >
          Abarrotes
        </h3>
        <p
          class="p-Productos"
        >
          Lorem ipsum dolor sit amet
        </p>
        <button
          class="VerMas"
        >
          Ver más
        </button>
      </div>
    </div>
    <div
      class="row-search"
    >
      <div
        class="cards"
      >
        <img
          alt="Cuidado"
          class="Cuidado"
          src="cuidadoPersonal.jpg"
        />
        <h3
          class="H3-Productos"
        >
          Cuidado
        </h3>
        <p
          class="p-Productos"
        >
          Lorem ipsum dolor sit amet
        </p>
        <button
          class="VerMas"
        >
          Ver más
        </button>
      </div>
    </div>
    <div
      class="row-search"
    >
      <div
        class="cards"
      >
        <img
          alt="Belleza"
          class="Belleza"
          src="belleza.png"
        />
        <h3
          class="H3-Productos"
        >
          Belleza
        </h3>
        <p
          class="p-Productos"
        >
          Lorem ipsum dolor sit amet
        </p>
        <button
          class="VerMas"
        >
          Ver más
        </button>
      </div>
    </div>
    <div
      class="row-search"
    >
      <div
        class="cards"
      >
        <img
          alt="Hogar"
          class="Hogar"
          src="hogar.jpg"
        />
        <h3
          class="H3-Productos"
        >
          Hogar
        </h3>
        <p
          class="p-Productos"
        >
          Lorem ipsum dolor sit amet
        </p>
        <button
          class="VerMas"
        >
          Ver más
        </button>
      </div>
    </div>
    <div
      class="row-search"
    >
      <div
        class="cards"
      >
        <img
          alt="Aseo"
          class="Aseo"
          src="aseoHogar.jpg"
        />
        <h3
          class="H3-Productos"
        >
          Aseo
        </h3>
        <p
          class="p-Productos"
        >
          Lorem ipsum dolor sit amet
        </p>
        <button
          class="VerMas"
        >
          Ver más
        </button>
      </div>
    </div>
  </div>
  <footer
    class="footer"
  >
    <img
      alt="Este es el logo de cheaper"
      class="NavBar-Logo-Home"
      src="logo.png"
    />
    <p
      class="copyright"
    >
       
      <span>
        ©
      </span>
       All rights reserved 2020
       
    </p>
  </footer>
</div>
`;
